src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC, forwardRef, memo, useCallback, useMemo, useRef } from 'react';
+import { FlatList, useWindowDimensions, ViewProps } from 'react-native';
+import { useComponentStyles, useMolecules } from '../../hooks';
+import type { ColumnProps } from '../Column';
+import Column from '../Column/Column';
+
+export type Props = ViewProps & {
+    data: ColumnProps[];
+    renderer: FC<{ name: string }>;
+};
+
+const keyExtractor = (item: ColumnProps) => item.name;
+
+function Grid({ data, style, renderer }: Props, ref: any) {
+    const { View } = useMolecules();
+    const { width } = useWindowDimensions();
+    const componentStyles = useComponentStyles('Grid', style);
+
+    const { gridStyles, rowStyles } = useMemo(() => {
```

Comment:
accept an optional breakpoint prop, if it is unavailable, take the breakpoints from the theme